plugins/plugin-shiki/src/node/types.ts

```diff
@@ -0,0 +1,149 @@
+import type {
+  BundledLanguage,
+  BundledTheme,
+  Highlighter,
+  LanguageInput,
+  ShikiTransformer,
+  SpecialLanguage,
+  StringLiteralUnion,
+  ThemeRegistration,
+  ThemeRegistrationRaw,
+} from 'shiki'
+
+export type ShikiLang =
+  | LanguageInput
+  | StringLiteralUnion<BundledLanguage>
+  | SpecialLanguage
+
+export type ShikiTheme =
+  | ThemeRegistration
+  | ThemeRegistrationRaw
+  | StringLiteralUnion<BundledTheme>
+
+/**
+ * Options of @vuepress/plugin-shiki
+ */
+export interface ShikiPluginOptions {
+  /**
+   * Languages to be loaded.
+   *
+   * Shiki does not preload any languages to avoid extra overhead. So you need
+   * to specify the languages you want to use.
+   *
+   * @see https://shiki.style/languages
+   */
+  langs?: ShikiLang[]
+
+  /**
+   * Fallback language when the specified language is not available.
+   */
+  defaultHighlightLang?: string
+
+  /**
+   * The single theme to use
+   *
+   * @see https://shiki.style/themes
+   */
+  theme?: ShikiTheme
+
+  /**
+   * The dark and light themes to use
+   *
+   * @see https://shiki.style/themes
+   */
+  themes?: {
+    dark: ShikiTheme
+    light: ShikiTheme
+  }
+
+  /**
+   * Setup Shiki instance
+   *
+   * Custom shiki setup function. You can customize the shiki instance by adding your own shikiSetup function in your config.
```

Comment:
This should be aligned with the official docs description.

I think for jsdocs,  we should better keep the description short and useful